server/graphql/core/src/loader/units_in_other_purchase_orders.rs

```diff
@@ -0,0 +1,58 @@
+use async_graphql::dataloader::*;
+use async_graphql::*;
+use repository::{
+    EqualFilter, PurchaseOrderFilter, PurchaseOrderLineFilter, PurchaseOrderLineRepository,
+    PurchaseOrderLineStatus, PurchaseOrderStatus, RepositoryError, StorageConnectionManager,
+};
+use std::collections::HashMap;
+
+pub struct UnitsInOtherPurchaseOrdersLoader {
+    pub connection_manager: StorageConnectionManager,
+}
+
+impl Loader<String> for UnitsInOtherPurchaseOrdersLoader {
+    type Value = f64;
+    type Error = RepositoryError;
+
+    async fn load(&self, ids: &[String]) -> Result<HashMap<String, Self::Value>, Self::Error> {
+        let connection = self.connection_manager.connection()?;
+        let repo = PurchaseOrderLineRepository::new(&connection);
+
+        let lines = repo.query_by_filter(
+            PurchaseOrderLineFilter::new().id(EqualFilter::equal_any(ids.to_vec())),
+        )?;
+
+        let mut result = HashMap::new();
+
+        for line in lines {
+            let line_row = &line.purchase_order_line_row;
+            let item_row = &line.item_row;
+
+            let other_confirmed_orders = repo.query_by_filter(
+                PurchaseOrderLineFilter::new()
+                    .item_id(EqualFilter::equal_to(&item_row.id))
+                    .purchase_order(
+                        PurchaseOrderFilter::new()
+                            .id(EqualFilter::not_equal_to(&line_row.purchase_order_id))
+                            .status(PurchaseOrderStatus::Confirmed.equal_to()),
+                    )
+                    .status(PurchaseOrderLineStatus::Sent.equal_to()),
+            )?;
+
+            // TODO: Reduce any other units received in GRs
+            let units_in_other_orders = other_confirmed_orders
```

Comment:
Did the summin' in the loader since we really only need to return 1 value... might change in the future if there are more stats we want to show